### PR TITLE
Switch to standard Django permissions, instead of using custom roles.

### PR DESCRIPTION
**Pull request by rebeccacremona**
As per https://github.com/harvard-lil/h2o/issues/1026, TODOs in the code, and a remark about Roles in https://github.com/harvard-lil/h2o/issues/872, this PR switches us over from custom permission-related code to Django's standard `is_superadmin`, `is_staff`, and `is_active`.

We set `is_active` to False by default, and toggle it to True when users verify their email addresses and set a password. We won't use it instead of deleting once-active users, as recommended by Django's default helptext for that field, so I have not included that helptext!

### CODE REVIEW

**Comment by rebeccacremona** [comment]
Hmm. I seem to have broken `pyflakes` linting.  Investigating.....

**Comment by rebeccacremona** [comment]
Ooh, terrible, `pyflakes` was borking on `from django.utils.translation import gettext_lazy as _`, which I took right out of the Django source code.... I don't need it, so just removed it, but not good!

File: web/main/models.py [hunk 1]
```diff
@@ -2431,18 +2401,19 @@ def related_resources(self):
...
-    # used to assign super_admin or case_admin status
-    roles = models.ManyToManyField(Role,
-        through=RolesUser
-    )
+    is_staff = models.BooleanField(default=False)
+    is_active = models.BooleanField(default=False)
+
+    # differentiate between real User model and AnonymousUser model:
+    is_authenticated = True
+    is_anonymous = False
```
Comment (rebeccacremona) [inline web/main/models.py]: @jcushman can you say a word about why we have to set `is_authenticated` and `is_anonymous` manually? I should be able to figure out, but I'm not getting it.

Comment (jcushman) [reply web/main/models.py]: Looks like we don't need to anymore! When those two lines were added we weren't inheriting from `AbstractBaseUser` yet.

Comment (rebeccacremona) [reply web/main/models.py]: Aha! Thanks!! I'll delete these lines too!

Comment (rebeccacremona) [reply web/main/models.py]: Deleted. And, corrected the reverse migration per discussion in Slack.

**Pull request merged**